pkg/chains/storage/grafeas/grafeas_test.go

```diff
@@ -13,486 +13,486 @@ limitations under the License.
-	// Tests will keep working if more methods are added in the future.
-	pb.UnimplementedGrafeasServer
-
-	// Assume there is only one project for storing notes and occurences
-	occurences map[string]*pb.Occurrence
-	notes      map[string]*pb.Note
-}
-
-func (s *mockGrafeasServer) CreateOccurrence(ctx context.Context, req *pb.CreateOccurrenceRequest) (*pb.Occurrence, error) {
-	if s.occurences == nil {
-		s.occurences = make(map[string]*pb.Occurrence)
-	}
-
-	occID := req.GetOccurrence().GetResourceUri()
-	expectedResponse := req.GetOccurrence()
-	expectedResponse.Name = occID // mock auto-generated id
-
-	s.occurences[occID] = expectedResponse
-	return expectedResponse, nil
-}
-
-func (s *mockGrafeasServer) CreateNote(ctx context.Context, req *pb.CreateNoteRequest) (*pb.Note, error) {
-	noteID := fmt.Sprintf("%s/notes/%s", req.GetParent(), req.GetNoteId())
-	expectedResponse := req.GetNote()
-	if s.notes == nil {
-		s.notes = make(map[string]*pb.Note)
-	}
-
-	if _, exists := s.notes[noteID]; exists {
-		return nil, gstatus.Error(codes.AlreadyExists, "note ID already exists")
-	}
-	s.notes[noteID] = expectedResponse
-	return expectedResponse, nil
-}
-
-func (s *mockGrafeasServer) ListOccurrences(ctx context.Context, req *pb.ListOccurrencesRequest) (*pb.ListOccurrencesResponse, error) {
-	// to make sure the occurrences we get are in order.
-	sortedOccurrencesInServer := []*pb.Occurrence{}
-	keys := []string{}
-	for k := range s.occurences {
-		keys = append(keys, k)
-	}
-	sort.Strings(keys)
-	for _, k := range keys {
-		sortedOccurrencesInServer = append(sortedOccurrencesInServer, s.occurences[k])
-	}
-
-	// if filter string is empty, the expected behaviour will be to return all.
-	if len(req.GetFilter()) == 0 {
-		return &pb.ListOccurrencesResponse{Occurrences: sortedOccurrencesInServer}, nil
-	}
-
-	// if the filter string is not empty, do the filtering.
-	// mock how uri filter works
-	uris := parseURIFilterString(req.GetFilter())
-
-	// result occurrences
-	occurrences := []*pb.Occurrence{}
-
-	for _, occ := range sortedOccurrencesInServer {
-		for _, uri := range uris {
-			if uri == occ.GetResourceUri() {
-				occurrences = append(occurrences, occ)
-			}
-		}
-	}
-
-	return &pb.ListOccurrencesResponse{Occurrences: occurrences}, nil
-}
-
-// parse a chained uri filter string to a list of uris
-// example:
-//   - input: `resourceUrl="foo" OR resourceUrl="bar"`
-//   - output: ["foo", "bar"]
-func parseURIFilterString(filter string) []string {
-	results := []string{}
-
-	// a raw filter string will look like `resourceUrl="foo" OR resourceUrl="bar"`
-	// 1. break them into separate statements
-	statements := strings.Split(filter, " OR ")
-
-	for _, statement := range statements {
-		// each statement looks like `resourceUrl="foo"`
-		// 2. extract `foo`
-		uriWithQuotes := strings.Split(statement, "=")[1]
-		uri := uriWithQuotes[1 : len(uriWithQuotes)-1]
-		results = append(results, uri)
-	}
-
-	return results
-}
+// import (
```

Comment:
Were tests going to done in a follow up PR?

Reply:
Yes! The tests will be done in this PR.

Reply:
The unit tests are finished. Please take a look. I also tested the backend with real grafeas server (container analysis). It works as expected. Thank you!